evaluate/report/csv.go

```diff
@@ -26,50 +27,109 @@ type CSVFormatter interface {
 	Rows() (rows [][]string)
 }
 
-// GenerateCSV returns the whole CSV as string.
-func GenerateCSV(formatter CSVFormatter) (csvData string, err error) {
-	var out strings.Builder
-	csv := csv.NewWriter(&out)
+// EvaluationRecord holds a line of the evaluation CSV.
+type EvaluationRecord struct {
+	// ModelID holds the model id.
+	ModelID string
+	// ModelName holds the model name.
+	ModelName string
+	// ModelCost holds the model cost.
+	ModelCost float64
 
-	if err := csv.Write(formatter.Header()); err != nil {
-		return "", pkgerrors.WithStack(err)
-	}
+	// LanguageID holds the language id.
+	LanguageID string
 
-	for _, row := range formatter.Rows() {
-		if err := csv.Write(row); err != nil {
-			return "", pkgerrors.WithStack(err)
+	// Assessments holds the assessments of an entry.
+	Assessments metrics.Assessments
+}
```

Comment:
This should probably also hold the task and repository, right?

Reply:
My 1st implementation held both the task and repository fields. The reasons I removed them are:
- I'm using the evaluation record to group the values and adding the assessments, so what would be e.g the `Repository` value when  we group two evaluation records from different repositories? 
- We don't need them for the current implementation

If we want to still keep them I guess we need another struct similar to `EvaluationRecord` but without the tasks and repositories, for grouping the values.

Reply:
Also, something we can do in a follow-up if needed

Reply:
Plz note down in the issue and resolve this.